.github/workflows/continuous-integration-stager.yml

```
@@ -0,0 +1,62 @@
+name: staged-continuous-integration
+
+on:
+  push:
+    branches:
+      - develop
+  pull_request:
+    paths-ignore:
+    - '**/*.md'
+    types: [ opened, reopened, synchronize ]
+
+concurrency:
+  group: ${ {github.event_name }}-${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{github.event_name == 'pull_request'}}
+
+permissions: read-all
+
+jobs:
+  # pre-stage
+  clang-format-check:
+    uses: ./.github/workflows/clang-format-check.yml
+
+  codeql:
+    # need to set manual permissions since it requires security-events write
+    # but then we have to set everything else manually to read since the
+    # default is overwritten
+    permissions:
+      # required for all workflows
+      security-events: write
+      # only required for workflows in private repositories
+      actions: read
+      contents: read
+
+    uses: ./.github/workflows/codeql.yml
+
+  initial-check:
+    uses: ./.github/workflows/continuous-integration-snifftest.yml
```

```suggestion
  sniff-test:
    uses: ./.github/workflows/continuous-integration-snifftest.yml
```
(for consistency, and because I find the name funny)

---

```suggestion
  initial-check:
    uses: ./.github/workflows/continuous-integration-smoketest.yml
```

---

I renamed it to smokiest